charms/kserve-controller/src/certs.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+
+import tempfile
+from pathlib import Path
+from subprocess import check_call
+
+SSL_CONFIG_FILE = "src/templates/ssl.conf.j2"
+
+
+def gen_certs(service_name: str, namespace: str, webhook_service: str):
+    """Generate certificates."""
+
+    ssl_conf = Path(SSL_CONFIG_FILE).read_text()
+    ssl_conf = ssl_conf.replace("{{ service_name }}", str(service_name))
+    ssl_conf = ssl_conf.replace("{{ namespace }}", str(namespace))
+    ssl_conf = ssl_conf.replace("{{ webhook_server_service }}", str(webhook_service))
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_path = Path(tmp_dir)
+        (tmp_path / "ssl.conf").write_text(ssl_conf)
+
+        # execute OpenSSL commands
+        check_call(["openssl", "genrsa", "-out", tmp_path / "ca.key", "2048"])
```

Comment:
Is there a python equivalent to these `openssl` calls?  I don't know much about this topic, but I'll be surprised if there isn't a python package that generates these outputs with an easier interface

Reply:
Maybe [pyopenssl](https://www.pyopenssl.org/en/latest/)?

Reply:
there is definitely, [cryptography](https://cryptography.io/en/latest/) is already utilized in the [tls-certificates-operator](https://github.com/canonical/tls-certificates-operator), we will be replacing our own certs implementation with `tls-certificate-operator` and `tls-certificates-interface`. I propose we first merge this, and then refactor with the tls operator.

Reply:
filed issue #104